typescript/src/ethereum.ts

```diff
@@ -244,4 +317,118 @@ export class Bridge implements ChainBridge {
       walletPublicKeyHash
     )
   }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainBridge#deposits}
+   */
+  async deposits(
+    depositTxHash: TransactionHash,
+    depositOutputIndex: number
+  ): Promise<RevealedDeposit> {
+    const depositKey = this.buildDepositKey(depositTxHash, depositOutputIndex)
+
+    const deposit = await this._bridge.deposits(depositKey)
+
+    return this.parseRevealedDeposit(deposit)
+  }
+
+  /**
+   * Builds the deposit key required to refer a revealed deposit.
+   * @param depositTxHash The revealed deposit transaction's hash.
+   * @param depositOutputIndex Index of the deposit transaction output that
+   *        funds the revealed deposit.
+   * @returns Revealed deposit data.
+   */
+  private buildDepositKey(
+    depositTxHash: TransactionHash,
+    depositOutputIndex: number
+  ): string {
+    const prefixedReversedDepositTxHash = `0x${Buffer.from(depositTxHash, "hex")
+      .reverse()
+      .toString("hex")}`
+
+    return utils.solidityKeccak256(
+      ["bytes32", "uint32"],
+      [prefixedReversedDepositTxHash, depositOutputIndex]
+    )
+  }
+
+  /**
+   * Parses a revealed deposit using data fetched from the on-chain contract.
+   * @param deposit Data of the revealed deposit.
+   * @returns Parsed revealed deposit.
+   */
+  private parseRevealedDeposit(deposit: any): RevealedDeposit {
+    return {
+      depositor: new Address(deposit.depositor),
+      amount: BigNumber.from(deposit.amount),
+      vault:
+        deposit.vault === constants.AddressZero
+          ? undefined
+          : new Address(deposit.vault),
+      revealedAt: BigNumber.from(deposit.revealedAt).toNumber(),
+      sweptAt: BigNumber.from(deposit.sweptAt).toNumber(),
+      treasuryFee: BigNumber.from(deposit.treasuryFee),
+    }
+  }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainBridge#activeWalletPublicKey}
+   */
+  async activeWalletPublicKey(): Promise<string | undefined> {
+    const activeWalletPubKeyHash = await this._bridge.activeWalletPubKeyHash()
+
+    if (
+      activeWalletPubKeyHash === "0x0000000000000000000000000000000000000000"
```

Comment:
Can we use `AddressZero` const from `ethers`?

Reply:
We could but `activeWalletPubKeyHash` is not an address. The only thing that is the same is the 20-byte length. I think it can be confusing if we use `AdressZero`.

Reply:
Ah, I see, you are right. Let's keep it as it is here 👍